src/Microsoft.DotNet.Darc/Darc/Operations/UpdateDependenciesOperation.cs

```diff
@@ -465,4 +471,119 @@ private static IEnumerable<DependencyDetail> GetDependenciesFromPackagesFolder(s
 
         return updatedDependencies;
     }
+
+    /// <summary>
+    /// Fetch subscription metadata and populate command options based on subscription settings.
+    /// This allows the subscription to be simulated using the existing update logic.
+    /// </summary>
+    private async Task PopulateOptionsFromSubscriptionAsync()
+    {
+        // Validate that subscription is not used with conflicting options
+        if (!string.IsNullOrEmpty(_options.Channel))
+        {
+            throw new DarcException("The --subscription parameter cannot be used with --channel. The subscription already specifies a channel.");
+        }
+
+        if (_options.BARBuildId > 0)
+        {
+            throw new DarcException("The --subscription parameter cannot be used with --id. The subscription determines which build to use.");
+        }
```

Comment:
I think this one is not correct. You might want to use everything from the subscription (excluded assets, etc.) but override the build ID.